File: Example/SwiftRuler/Rules/CustomRule.m
//
//  CustomRule.m
//  SwiftRuler
//
//  Created by Rawipon Srivibha  on 8/9/17.
//  Copyright © 2017 Rawipon Srivibha. All rights reserved.
//

#import "CustomRule.h"

@implementation CustomRule

- (BOOL)validateWithCandidate:(id)candidate {
  if(![candidate isKindOfClass:[NSString class]]) {
    return NO;
  }
  
  NSString *candidateString = (NSString*)candidate;
  NSCharacterSet *capSet = [NSCharacterSet characterSetWithCharactersInString:@"ABCDEFGHIJKLMNOPQRSTUVWXYZ"];
  if([candidateString rangeOfCharacterFromSet:capSet].location != NSNotFound) {
    return NO;
  }
  
  return YES;
}

@end

File: Example/SwiftRuler/Rules/CustomRule.swift
//
//  CustomRule.swift
//  SwiftRuler_Example
//
//  Created by Rawipon Srivibha on 6/6/19.
//  Copyright © 2019 Rawipon. All rights reserved.
//

import Foundation
import SwiftRuler

class CustomRule: Rule {
  
  func validate(candidate: Any?) -> Bool {
    guard let stringCandidate = candidate as? String else {
      return false
    }
    
    return stringCandidate.rangeOfCharacter(from: CharacterSet(charactersIn: "ABCDEFGHIJKLMNOPQRSTUVWXYZ")) != nil
  }
  
}
